engine/dlib/src/dlib/sys_cocoa.mm

```diff
@@ -33,9 +33,14 @@
 {
     static inline void PatchSystemName(char *system_name, uint32_t buffer_size)
     {
+        // iPads with iOS version 15+ have started reporting iPadOS
+        if (dmStrCaseCmp(system_name, "iPadOS")==0)
```

Comment:
If the string should always be the same thing, then why not hardcode it?
Why add extra runtime code?

```
const char* GetSystemName() {
#if defined(__arm__) || defined(__arm64__) || defined(IOS_SIMULATOR)
    const char* system_name = "iPhone OS";
#else
    const char* system_name = "Darwin";
#endif
    return system_name;
}

dmStrlCpy(info->m_SystemName, GetSystemName(), sizeof(info->m_SystemName));
```

```

Reply:
Good point. I will hardcode "iPhone OS".